### PR TITLE
feat: add summary to conversations model

### PR DESCRIPTION
**Pull request by scmmishra**
> ~🚨 Merging this to a feature branch, coz the summary format update is a breaking change, and will break in the UI if it is not merged without the frontend changes~
> updated the prompt, this can be used in the main branch

This PR updates conversations model, to save summary and its timestamp. This PR has the following changes

1. Add the `summary` and `summary_generated_at` fields to summary model
2. Save summary once generated in `openai/processor_service.rb`
3. Add the fields to `partials/_conversation.json.jbuilder`
4. Cleanup summary markdown before rendering it

## Type of change

- [x] New feature (non-breaking change which adds functionality)

### CODE REVIEW

File: enterprise/lib/enterprise/integrations/openai_prompts/summary.txt
Comment (scmmishra) [inline enterprise/lib/enterprise/integrations/openai_prompts/summary.txt:1]: Converting this to markdown to better parse it

File: db/schema.rb [hunk 1]
```diff
@@ -449,8 +449,10 @@
     t.datetime "assignee_last_seen_at", precision: nil
     t.datetime "first_reply_created_at", precision: nil
     t.integer "priority"
-    t.bigint "sla_policy_id"
     t.datetime "waiting_since"
+    t.bigint "sla_policy_id"
```
Comment (sojan-official) [inline db/schema.rb:453]: Order change of 'sla_policy_id' intended?

File: db/schema.rb [hunk 2]
```diff
@@ -615,6 +617,19 @@
...
+    t.bigint "labelable_id", null: false
+    t.datetime "created_at", null: false
+    t.datetime "updated_at", null: false
+    t.index ["label_id"], name: "index_label_associations_on_label_id"
+    t.index ["labelable_type", "labelable_id"], name: "index_label_associations_on_contact", where: "((labelable_type)::text = 'Contact'::text)"
+    t.index ["labelable_type", "labelable_id"], name: "index_label_associations_on_conversation", where: "((labelable_type)::text = 'Conversation'::text)"
+    t.index ["labelable_type", "labelable_id"], name: "index_label_associations_on_labelable"
+    t.index ["labelable_type", "labelable_id"], name: "index_label_associations_on_labelable_type_and_labelable_id"
+  end
+
```
Comment (sojan-official) [inline db/schema.rb:632]: Are the changes in schema.rb from the migrations in this PR alone?

**Review by sojan-official — COMMENTED** [review]
The code looks good to me. 
Please double-check the schema.rb

File: db/schema.rb (hunk 2 above)
Comment (scmmishra) [reply db/schema.rb:632]: Oh these shouldn't be here, not sure how these came in, fixing it

**Comment by scmmishra** [comment]
@sojan-official Done

**Comment by scmmishra** [comment]
@sojan-official this PR is good to go on my end, PTAL

**Comment by sojan-official** [comment]
@scmmishra  Let's reopen this once we have a clearer direction on how we will take the `summary` feature forward. Let's keep the model attributes trim so that it's easier to document and work with.  In the cases where the attribute is optional  / doesn't require indexing, let's leverage the content attributes jsonb field as well.

**Comment by scmmishra** [comment]
> @scmmishra  Let's reopen this once we have a clearer direction on how we will take the `summary` feature forward. Let's keep the model attributes trim so that it's easier to document and work with.  In the cases where the attribute is optional  / doesn't require indexing, let's leverage the content attributes jsonb field as well. 


Yeah sounds good to me

**Pull request closed without merging**